### DISCUSSION TITLE
What is the judgement of the value term.buffer.active.type based on?

### DISCUSSION THREAD

**Discussion by alanhe421** [sourcelair/xterm.js#4594] [Q&A]
What is the judgement of the value term.buffer.active.type based on? I encountered a problem. Even in the case of editing in VI, the return value of term.buffer.active.type is still normal. I don't know how to proceed.

## version  info
```
 "xterm": "5.1.0",
    "xterm-addon-attach": "^0.8.0",
    "xterm-addon-canvas": "^0.4.0",
    "xterm-addon-fit": "^0.7.0",
    "xterm-addon-search": "^0.11.0",
    "xterm-addon-serialize": "^0.9.0",
    "xterm-addon-unicode11": "^0.5.0",
    "xterm-addon-web-links": "^0.8.0",
    "xterm-addon-webgl": "^0.14.0",
```

**Comment by Tyriar** [comment]
I don't think there's any issue with that API. Here's the implementation:

https://github.com/xtermjs/xterm.js/blob/f79e9a1951c2f9fca8f5e1ba735555a1e7d4cb74/src/common/public/BufferNamespaceApi.ts#L18-L27

If you're on Windows it might not be what you expect due to pty emulation.
